src/test/netbase_tests.cpp

```diff
@@ -395,6 +395,51 @@ BOOST_AUTO_TEST_CASE(netpermissions_test)
     BOOST_CHECK(NetWhitebindPermissions::TryParse("@1.2.3.4:32", whitebindPermissions, error));
     BOOST_CHECK_EQUAL(whitebindPermissions.m_flags, PF_NONE);
 
+    // "noban" implies "download"
+    BOOST_CHECK(NetWhitebindPermissions::TryParse("noban@1.2.3.4:32", whitebindPermissions, error));
+    BOOST_CHECK_EQUAL(whitebindPermissions.m_flags, PF_NOBAN);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_DOWNLOAD);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_NOBAN);
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_DOWNLOAD));
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_NOBAN));
+
+    NetPermissions::ClearFlag(whitebindPermissions.m_flags, PF_DOWNLOAD);
```

Comment:
out of scope of this PR: this `ClearFlag()` will leave the flags in an invalid state `0b10000` that does not equal any of the `PF_*` flags or a combination of them. This shouldn't be even possible/allowed.

It is like having some fruits in a bag - apple, orange, pear and when you remove the apple, then the bag does not contain fruits anymore, it now contains a frog :frog:.

Reply:
Good point. Added an assert and doxygen doc explanation to `ClearFlags` to allow only clearing only the implicit flag, as it's the only current use.